tests/test_util.py

```diff
@@ -222,12 +221,19 @@ async def asyncSetUp(self):
     async def asyncTearDown(self):
         await self.http.close()
 
+    async def test_correct_content_type(self):
+        await get_extra_data_info_from_url(
+            url=f"https://ftp.gnu.org/gnu/gzip/gzip-1.12.tar.gz",
```

Comment:
For a software distribution site, wrong content type is clearly a bug that can (and should) be fixed some day and break out test. Instead, use some URL that reliably returns invalid content type (e.g. `https://httpbingo.org/response-headers?Content-Type=application/x-tar`), and verify that the new check raises an error.

Reply:
I think detection via magic number doesn't work with httpbingo:

alexander@PC:~/Downloads> file gzip-1.12.tar.gz 
gzip-1.12.tar.gz: gzip compressed data, max compression, from Unix, original size modulo 2^32 6041600
alexander@PC:~/Downloads> file response-headers\?Content-Type\=application%2Fx-tar 
response-headers?Content-Type=application%2Fx-tar: JSON text data

In the test case I'm checking against an incorrect mime type instead.

Reply:
Ah, now I understand what you meant. That the httpbingo response has a mismatch between the actual content type as determined by `file` and the `content-type` field in the http header of the response, right?

Reply:
In urlchecker.py, get_extra_data_info_from_url() is called without the content_type_deny parameter, and application/gzip is not among the WRONG_CONTENT_TYPES_ in src/lib/init.py, so content_type_rejected() would return false for https://httpbingo.org/response-headers?Content-Type=application/gzip and the magic number wouldn't be checked

Aren't these two disparate aspects? 1) getting the content type. This has in the past been done by checking the http response header. With my MR, it'd be determined using the magic number 2) rejecting URLs if the content type matches one of those in the deny list.
So I'd imagine the proper way would be to always get the content_type from the magic number of the first chunk and cancel the download if content_type_rejected(actual_content_type) returns true